### PR TITLE
[12.0][FIX][WIP] mrp.wizard.change_production_qty sub bom row delete

### PR DESCRIPTION
**Pull request by mymage**
This is a proposal to solve an issue highlighted #57702 wich evidences the lack of upload of a MO BOM when a row of a sub-BOM is deleted.

Current behavior before PR:
On a MO of a product with a sub-bom of type kit, when a row of the sub-bom is set to quantity 0, the relative row on the MO is deleted; when the the row of a sub-bom is deleted, the relative row on the MO remain inalterated.

Desired behavior after PR is merged:
On a MO of a product with a sub-bom of type kit, when a row of the sub-bom is set to quantity 0 or is deleted, the relative row on the MO is deleted.

[WIP] (thanks to those who will help)
Add tests

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/wizard/change_production_qty.py [hunk 1]
```diff
@@ -46,6 +46,12 @@ def change_prod_qty(self):
         for wizard in self:
             production = wizard.mo_id
             produced = sum(production.move_finished_ids.filtered(lambda m: m.product_id == production.product_id).mapped('quantity_done'))
+
+            # memorize initial MO bom movement rows not added manually
+            mo_ini_bom_move_ids = []
+            for move_id in self.mapped("mo_id.move_raw_ids").filtered(lambda x: x.origin):
+                mo_ini_bom_move_ids.append(move_id)
+
```
Comment (sergiocorato) [inline addons/mrp/wizard/change_production_qty.py:54]:
```suggestion
            mo_ini_bom_move_ids = self.mapped("mo_id.move_raw_ids").filtered(lambda x: x.origin)

```

File: addons/mrp/wizard/change_production_qty.py [hunk 2]
```diff
@@ -74,6 +83,13 @@ def change_prod_qty(self):
 
                 production._update_raw_move(line, line_data)
 
+            # delete initial MO bom rows not present in actual bom
+            if self.mo_id.state not in ('progress','done','cancel'):
```
Comment (sergiocorato) [inline addons/mrp/wizard/change_production_qty.py:87]:
```suggestion
            if production.state not in ('progress','done','cancel'):
```

File: addons/mrp/wizard/change_production_qty.py [hunk 3]
```diff
@@ -74,6 +83,13 @@ def change_prod_qty(self):
...
+                for mo_ini_move_id in mo_ini_bom_move_ids:
+                    if mo_ini_move_id not in mo_act_bom_move_ids:
+                        mo_ini_move_id.state = 'draft'
+                        mo_ini_move_id.unlink()
```
Comment (sergiocorato) [inline addons/mrp/wizard/change_production_qty.py:91]:
```suggestion
                to_unlink_move_ids = [
                    x for x in mo_ini_bom_move_ids if x not in mo_act_bom_move_ids]
                to_unlink_move_ids.write({
                    'state': 'draft'
                })
                to_unlink_move_ids.unlink()
```

**Comment by C3POdoo** [comment]
Dear @mymage,

Thank you for your contribution but the version 12.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**